src/app/loadout/ingame/ingame-loadout-utils.ts

```diff
@@ -112,6 +115,28 @@ export function implementsDimLoadout(
     }
     gameLoadoutMods.splice(pos, 1);
   }
+
+  // Ensure that the dimsubclass abilities, aspect and fragments are accounted for
+  // so that builds using the same subclass but different setups are identified.
+  const dimSubclass = dimResolvedLoadoutItems.find(
+    (rli) => rli.item.bucket.hash === BucketHashes.Subclass
+  );
+  if (dimSubclass?.loadoutItem?.socketOverrides) {
+    // This was checked as part of item matching.
+    const inGameSubclass = inGameLoadout.items.find(
+      (item) => item.itemInstanceId === dimSubclass.loadoutItem.id
+    )!;
+
+    const dimSubclassPlugs = defs ? getSubclassPlugs(defs, dimSubclass) : [];
```

Comment:
nnnno way should implementsDimLoadout be running if defs aren't loaded right? maybe we should chort circuit inGameLoadoutsWithMetadataSelector  if there's no defs

Reply:
The way i see it is dim basically doesnt work if defs arent loaded. This is just a formality as i don't really like asserting not null.

Reply:
that's fair just a difference of how high it belongs

Reply:
Yeah you are right, I'll move it up. I'm going to have a pass at removing this awkwardness I think

Reply:
Hmmm we are handling undefined much lower in the mods case too, see [this](https://github.com/DestinyItemManager/DIM/blob/2a8bafecaf137b8f493fbe68bc3c6baa9caebbbe/src/app/loadout/ingame/selectors.ts#L71-L79) which points to the `getModsFromLoadout` function.

I might move this up and then also change that to expect defs to be loaded.

Reply:
Done